deployment/HDI-Spark/Notebooks/deploy-sparkdotnet-repl.sh

```diff
@@ -0,0 +1,116 @@
+#!/bin/bash
+#
+# Install Livy and SparkMagic on Head node
+#
+
+if  [[ $HOSTNAME == hn* ]] ;
+then
+    # Update Livy Jars
+    sudo wget https://ruinliureplhdistorage.blob.core.windows.net/ruinliurepl/livy_jar.zip
+    sudo unzip livy_jar.zip
+
+    sudo cp -rf livy_jar/* /usr/hdp/current/livy2-server/
+
+    # Update SparkMagic
+    sudo wget https://ruinliureplhdistorage.blob.core.windows.net/ruinliurepl/sparkmagic.zip
+    sudo unzip sparkmagic.zip
+    
+    if [ ! -d "/usr/bin/anaconda/lib/python2.7/site-packages/sparkmagic/kernels/sparkdotnetkernel" ]
+    then
+        sudo mkdir /usr/bin/anaconda/lib/python2.7/site-packages/sparkmagic/kernels/sparkdotnetkernel
+    fi
+
+    sudo cp -f sparkmagic/config.json /home/spark/.sparkmagic/
+    sudo cp -rf sparkmagic/* /usr/bin/anaconda/lib/python2.7/site-packages/sparkmagic/
+
+    # install sparkdotnetkernel
+    sudo /usr/bin/anaconda/bin/jupyter-kernelspec install /usr/bin/anaconda/lib/python2.7/site-packages/sparkmagic/kernels/sparkdotnetkernel
+
+    # Remove the temporary livy jars and sparkmagic files.
+    sudo rm livy_jar.zip
+    sudo rm -rf livy_jar
+    sudo rm sparkmagic.zip
+    sudo rm -rf sparkmagic
+fi
+
+#
+# Install SparkDotnet and Worker on Worker node
+#
+
+if  [[ $HOSTNAME == wn* ]] ;
+then
+    set +e
+
+    # Uncomment if you want full tracing (for debugging purposes)
+    #set -o xtrace
+
+    # Install SparkDotNet
+    SPARK_DOTNET_VERSION=$1
+
+    sudo dpkg --purge --force-all packages-microsoft-prod
+    sudo wget -q https://packages.microsoft.com/config/ubuntu/`lsb_release -rs`/packages-microsoft-prod.deb -O packages-microsoft-prod.deb
+    sudo dpkg -i packages-microsoft-prod.deb
+    sudo add-apt-repository universe
+    sudo apt-get -yq install apt-transport-https
+    sudo apt-get -yq update
+    sudo apt-get -yq install dotnet-sdk-3.0
+
+    sudo dotnet tool install dotnet-try --add-source https://dotnet.myget.org/F/dotnet-try/api/v3/index.json --tool-path /usr/share/dotnet-tools --version 1.0.19473.13
+
+    # copy .NET for Apache Spark jar to SPARK's jar folder
+    sudo wget "https://ruinliureplhdistorage.blob.core.windows.net/ruinliurepl/microsoft-spark-2.4.x-${SPARK_DOTNET_VERSION}.jar" -O "microsoft-spark-2.4.x-${SPARK_DOTNET_VERSION}.jar"
```

Comment:
are you planning to adding the `.jar` file here every release ?  As we discussed earlier, why not have it download the `.nupkg` and extract the jar from there.

Reply:
Will do

Reply:
A `.nupkg` is essentially a `zip` file.  You can use `unzip` to extract the contents.

Reply:
Changed